pkg/sql/schemachanger/scplan/plan.go

```diff
@@ -108,14 +108,35 @@ func MakePlan(ctx context.Context, initial scpb.CurrentState, params Params) (p
 func makePlan(ctx context.Context, p *Plan) (err error) {
 	{
 		start := timeutil.Now()
+		var oldCurrent []scpb.Status
+		if p.Params.ExecutionPhase <= scop.PreCommitPhase {
+			// We need the full graph at this point because the plan may transition
+			// back to the initial statuses at pre-commit time.
+			oldCurrent = append(oldCurrent, p.CurrentState.Current...)
```

Comment:
nit: `make` the slice of the correct size here before the append. No reason not to. Append will not do what you want

Reply:
I'm not particularly concerned about slightly over-allocating here, but fine.

> Append will not do what you want

I don't understand, please explain.

Reply:
Fine.